src/shared/plugins/cookie-message/routes.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const Joi = require('@hapi/joi');
+const { createRoutePair } = require('shared/lib/route-helpers');
+const controller = require('./controller');
+
+module.exports = {
+  ...createRoutePair(controller, 'cookies', {
+    path: '/cookies',
+    config: {
+      auth: {
+        strategy: 'standard',
+        mode: 'try'
+      },
+      description: 'View cookie details and set choices',
+      plugins: {
+        viewContext: {
+          pageTitle: 'Cookies'
+        }
+      },
+      validate: {
+        query: Joi.object({
+          redirectPath: Joi.string().max(256).optional(),
+          form: Joi.string().guid().optional()
+        })
+      }
+    }
+  }),
+  getSetCookiePreferences: {
+    path: '/set-cookie-preferences',
+    method: 'get',
+    handler: controller.getSetCookiePreferences,
+    config: {
+      auth: {
+        strategy: 'standard',
+        mode: 'try'
+      },
+      description: 'View cookie details and set choices',
+      validate: {
+        query: Joi.object({
+          acceptAnalytics: Joi.boolean().truthy('1').falsy('0'),
```

Comment:
Just purely out of interest, can't we just take actual booleans here? E.g. `true` or `false` rather than `1` vs `0`?

Reply:
As they are decoded query params they will always be strings.  But we could do:
```
Joi.boolean().truthy('true').falsy('false'),
```
And then the URL would be /blah?acceptAnalytics=true
```
Do you think that is neater?